synapse/storage/databases/main/devices.py

```diff
@@ -1772,9 +1783,14 @@ async def get_uncoverted_outbound_room_pokes(
             LIMIT ?
         """
 
-        def get_uncoverted_outbound_room_pokes_txn(txn):
+        def get_uncoverted_outbound_room_pokes_txn(
+            txn: LoggingTransaction,
+        ) -> List[Tuple[str, str, str, int, Optional[Dict[str, str]]]]:
```

Comment:
I'm not sure about the last entry. Isn't `opentracing_context` a nullable text column in the DB? I.e. `Optional[str]`

Reply:
I was also a little bit confused. It is inserted as `Dict[str, str]`.
https://github.com/matrix-org/synapse/blob/f5668f0b4a6cca659ae98d3cb3714692ba488e89/synapse/storage/databases/main/devices.py#L1668-L1687

Reply:
On line 1687 the `context` is encoded to a string, and that's what we put in the DB on line 1710.

Reply:
I think there might have been some mistakes in https://github.com/matrix-org/synapse/pull/12321/files#r855043253

Reply:
I have added a conversion from `str` to `Dict`. This should be better.

Reply:
I think this means we decode to get a Dict... only to reencode it here? https://github.com/matrix-org/synapse/blob/5125fbc87146a0e699e6a03af84cab20d0c1693b/synapse/storage/databases/main/devices.py#L1698

I.e. I'm not sure if we ever use the context as a dict. @erikjohnston does this change seem sensible to you (in the light of #12321)?

Reply:
See also #12555.